graylog2-server/src/main/java/org/graylog2/migrations/V202201281638_GeoIpResolverConfigMigration.java

```diff
@@ -66,6 +84,12 @@ public ZonedDateTime createdAt() {
     @Override
     public void upgrade() {
 
+        MigrationCompletion completion = clusterConfigService.get(MigrationCompletion.class);
+        if (completion != null) {
+            LOG.info("Migration was already completed on '{}'", completion.completionDate());
```

Comment:
This works as expected to avoid running the migration multiple times. 👍  Nicely done. 

Can the log statement be changed to `debug` level? Many migrations are being skipped in this way after only running once. The pattern is usually to silently skip them, to avoid many "migration completed" log entries on server start up time, which could be distracting for the user.

I like the idea of capturing the date/time the migration was run, but the migration completion object in Mongo already contains the "last_updated" date, which is the same as the date when the migration has run. If the migration date ever needed to be identified, Mongo could be queried to find it. I think the payload `completion_date` can probably be removed since it seems redundant (and I don't think it's that important to include in the debug log statement). What do you think?
```
{
        "_id" : ObjectId("61f92d22bc3f6cb2a9369c51"),
        "type" : "org.graylog2.migrations.V202201281638_GeoIpResolverConfigMigration.MigrationCompletion",
        "payload" : {
                "completion_date" : ISODate("2022-02-01T12:52:50.437Z")
        },
        "last_updated" : ISODate("2022-02-01T12:52:50.438Z"),
        "last_updated_by" : "e065896b-8a9a-4f45-83f2-e740525ed035"
}
```

How about capturing the original GeoIp configuration and the resulting configuration after running the migration in the MigrationCompleted object instead? That would provide some audit history for which operations the migration actually completed. If that requires too much work (looks like the migration currently only performs the set operations without a query), I also think it's completely fine to write an empty MigrationCompleted marker object (with no fields).

Reply:
Updated log level to DEBUG and removed fields from migration status entity class.